tests/acceptance/rfc268-actions-test.js

```diff
@@ -63,6 +67,38 @@ module('Acceptance | actions [rfc268]', function(hooks) {
     assert.equal(page.screen, '15');
   });
 
+  test('allows testing loading behavior using returned promise', async function(assert) {
+    await page.visit();
+    await page.keys.clickOn('1');
+    await page.keys.clickOn('2');
+    await page.keys.sum();
+    await page.keys.clickOn('3');
+    let promise = page.keys.asyncEqual();
+
+    await waitUntil(() => page.isLoading);
+
+    await promise;
+
+    assert.notOk(page.isLoading);
+    assert.equal(page.screen, '15');
+  });
+
+  test('allows testing loading behavior using settled()', async function(assert) {
+    await page.visit();
+    await page.keys.clickOn('1');
+    await page.keys.clickOn('2');
+    await page.keys.sum();
+    await page.keys.clickOn('3');
+    page.keys.asyncEqual();
+
+    await waitUntil(() => page.isLoading);
```

Comment:
Yeah, I thought about that too. But then I realized that probably sometimes you want to wait for an element to exist, and sometimes wait for it to be visible. (`@ember/test-helpers`'s `waitFor` just waits for it to exist). So then I thought maybe an optional callback to verify conditions other than just pure existence...then I thought that doesn't seem to really add any value beyond using `waitUntil()` directly:

```javascript
await waitUntil(() => page.foo);
```

or

```javascript
await waitUntil(() => page.isFooVisible);
```

So it's fully doable, but I couldn't convince myself it would add much value...

Reply:
I think the most common use case is to wait for an element visibility. 
So `waitFor` property could always wait for `isVisible` of the node. If the user wants to customize a wait condition he can always use a new good `waitUntil` helper.

However it's totally unrelated to the PR. I've just realized that it can be a good addition to API.